go.mod

```diff
@@ -39,7 +39,7 @@ require (
 	golang.org/x/sync v0.11.0
 	golang.org/x/time v0.5.0
 	google.golang.org/grpc v1.63.2
-	google.golang.org/protobuf v1.33.0
+	google.golang.org/protobuf v1.36.1
```

Comment:
Ideally this change would not be part of this PR.  Instead, ideally, we would wait for https://github.com/elastic/fleet-server/pull/4702 to be merged into `9.0` first, before (rebasing and) merging this PR here.